### PR TITLE
add 2D buffer support

### PR DESCRIPTION
**Pull request by xfengcarl**
standard usage: permb  control buffer, it is always a 2D buffer related with MB Postion

Signed-off-by: Carl.Zhang <carl.zhang@intel.com>

### CODE REVIEW

File: va/va.h [hunk 1]
```diff
@@ -2180,6 +2180,30 @@ VAStatus vaCreateBuffer (
...
+ */
+VAStatus vaCreateBuffer2D (
+    VADisplay dpy,
+    VAContextID context,
+    VABufferType type,
+    unsigned int width,
+    unsigned int height,
+    VABuffer2D *buf2d
+);
+
```
Comment (xhaihao) [inline va/va.h]: Are all VABufferType types acceptable?

File: va/va_backend.h [hunk 2]
```diff
@@ -140,7 +140,9 @@ struct VADriverVTable
 		unsigned int size,		/* in */
 		unsigned int num_elements,	/* in */
 		void *data,			/* in */
-		VABufferID *buf_id		/* out */
+                unsigned int *unit_size,
+                unsigned int *pitch,
+                VABufferID *buf_id
```
Comment (fhvwy) [inline va/va_backend.h]: Please don't make this an incompatible driver API change by modifying the signature of the existing function - instead add the new function vaCreateBuffer2() in the driver API just like the external API.

File: va/va.h [hunk 3]
```diff
@@ -2181,6 +2181,24 @@ VAStatus vaCreateBuffer (
...
+ * can call vaMapBuffer with this Buffer ID to get virtual address. 
+ */
+VAStatus vaCreateBuffer2(
+    VADisplay dpy,
+    VAContextID context,
+    VABufferType type,
+    unsigned int width,
+    unsigned int height,
+    unsigned int *unit_size,
+    unsigned int *pitch,
```
Comment (fhvwy) [inline va/va.h]: I think this needs a clearer description of what it does.  Is it that you supply width and height, and based on those and the buffer type the values pointed to by unit_size and pitch are written with something which you need to know when mapping the buffer?  (An example might help.)

Comment (xfengcarl) [reply va/va.h]: yes, when you create 2D buffer, you certainly need the pitch for the alignment information, unit_size is another one, for example, we need 4 bytes to represent something of one MB, the unit_size should be 4

File: va/va_backend.h (hunk 2 above)
Comment (xfengcarl) [reply va/va_backend.h]: add another function pointer in the v-table , right?

Comment (xhaihao) [reply va/va_backend.h]: Yes, you should add another function pointer at the end of  the vtable.  your change breaks the compiling of other vaapi backend drivers

File: va/va.h (hunk 3 above)
Comment (xhaihao) [reply va/va.h]: Agree with @fhvwy, could you update the comment?

File: va/va_backend.h (hunk 2 above)
Comment (xfengcarl) [reply va/va_backend.h]: accept

File: va/va.h (hunk 3 above)
Comment (xfengcarl) [reply va/va.h]: accept, will add a example

**Pull request merged**